htsinfer/infer_count_motifs.py

```diff
@@ -0,0 +1,91 @@
+"""Infer motif frequency from sample data."""
+
+
+from typing import Dict
+
+
+def count_motifs(input_sequences: list, min_motif_length: int,
+                 max_motif_length: int, nucleic_acid: str) -> dict:
+    """Function that calculates the occurrence of all motifs in one or
+    multiple sequences and returns a dictionary with all motifs within
+    the specified length and their occurrence.
+
+        Args:
+            input_sequences (list): list of sequences
+            min_motif_length (int): minimal length of motif
+            max_motif_length (int): maximal length of motif
+            nucleic_acid (string): type of nucleic acid (dna, rna)
+
+        Returns:
+            sorted_dict (dict): paired motif frequency {motif: frequency }
+
+        Raises:
+            TypeError: input_sequences is not a list
+            TypeError: min_motif_length or max_motif_length is not an integer
+            ValueError: nucleic_acid is not in specified range
+            ValueError: max_motif_length is longer than the shortest list
+    """
+
+    if not isinstance(input_sequences, list):
+        raise TypeError('input_sequences is not a list of sequences!')
+
+    if not input_sequences:
+        raise TypeError('input_sequences is not a list of sequences!')
+
+    if not isinstance(min_motif_length, int):
+        raise TypeError('min_motif_length is not an integer!')
+
+    if not isinstance(max_motif_length, int):
+        raise TypeError('max_motif_length is not an integer!')
+
+    if nucleic_acid.lower() not in ['rna', 'dna']:
+        raise ValueError('nucleic_acid is not in specified range (dna/rna)!')
+
+    if max_motif_length > len(min(input_sequences)):
+        raise ValueError('max_motif_length is longer than the shortest list')
```

Comment:
Didn't you want to write:
> is longer than the shortest ~list~ _sequence_
?

Still, I don't really get why this is a problem. Shouldn't you make sure the _minimal_ motif length is not bigger than the shortest sequence? I think it's fine if the maximum motif length is never found